controllers/operatorscaler_controller.go

```diff
@@ -0,0 +1,333 @@
+func (r *OperatorScalerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	r.ctx = ctx
+	r.log = log.FromContext(ctx)
+
+	r.log.Info("starting reconcile")
+
+	if err := r.isOdfDependenciesCsvReady(); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := r.reconcileOperators(); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := r.reconcileDynamicWatchers(req.Name); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	r.log.Info("ending reconcile")
+	return ctrl.Result{}, nil
+}
+
+func (r *OperatorScalerReconciler) isOdfDependenciesCsvReady() error {
+
+	odfDepsCsv := &oprv1a1.ClusterServiceVersion{}
+
+	err := r.Client.Get(r.ctx, types.NamespacedName{Name: OdfDepsSubscriptionStartingCSV, Namespace: r.OperatorNamespace}, odfDepsCsv)
+	if err != nil {
+		r.log.Error(err, "failed getting odf-deps csv", "csvName", OdfDepsSubscriptionStartingCSV)
+		return err
+	}
+
+	if odfDepsCsv.Status.Phase != oprv1a1.CSVPhaseSucceeded {
+		err = fmt.Errorf("csv %s is not in succeeded state", OdfDepsSubscriptionStartingCSV)
+		r.log.Error(err, "waiting for csv to be in succeeded state")
+		return err
+	}
+
+	return nil
+}
+
+func (r *OperatorScalerReconciler) reconcileOperators() error {
+
+	var returnErr error
+
+	for _, metaData := range MetaDataList {
+
+		crList := &metav1.PartialObjectMetadataList{}
+		crList.TypeMeta.APIVersion = metaData.ApiVersion
+		crList.TypeMeta.Kind = metaData.Kind
+
+		err := r.Client.List(r.ctx, crList)
+		if err != nil {
+			if !meta.IsNoMatchError(err) {
+				msg := fmt.Sprintf("failed listing %s", metaData.Kind)
+				r.log.Error(err, msg)
+				multierr.AppendInto(&returnErr, err)
+			}
+		} else {
+			if len(crList.Items) >= 0 {
+
+				for _, pkgName := range metaData.PkgNames {
+					key := fmt.Sprintf(CsvLabelKey, pkgName, r.OperatorNamespace)
+
+					csvList := &oprv1a1.ClusterServiceVersionList{}
+					err = r.Client.List(
+						r.ctx, csvList,
+						client.InNamespace(r.OperatorNamespace),
+						client.MatchingLabels(map[string]string{key: ""}),
+					)
+					if err != nil {
+						r.log.Error(err, "failed listing csvs with label", "label", key)
+						multierr.AppendInto(&returnErr, err)
+					} else {
+						err = r.updateCsvsReplicas(csvList.Items...)
+						if err != nil {
+							r.log.Error(err, "failed updating csvs replica")
+							multierr.AppendInto(&returnErr, err)
+						}
+					}
+				}
+			}
+		}
+	}
+
+	return returnErr
+}
+
+func (r *OperatorScalerReconciler) updateCsvsReplicas(csvList ...oprv1a1.ClusterServiceVersion) error {
+
+	var returnErr error
+
+	for idx := range csvList {
+		csv := &csvList[idx]
+		var updateRequired bool
+		for i := range csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs {
+			if *csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[i].Spec.Replicas != int32(1) {
+				updateRequired = true
+			}
+			csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[i].Spec.Replicas = ptr.To(int32(1))
```

Comment:
1. Lets use a ref to the spec to make this more readable
2. use less than instead of not equal
```suggestion
			spec =  &csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[i].Spec
			if *spec.Replicas < int32(1) {
				spec.Replicas = ptr.To(int32(1))
				updateRequired = true
			}
			
```